google/oauth2/service_account.py

```diff
@@ -448,6 +487,14 @@ def refresh(self, request):
             self.token = access_token
             self.expiry = expiry
 
+        # In case of trust boundary never been fetched.
+        if (
+            self._trust_boundary is None
+            and self.universe_domain in _client.TRUST_BOUNDARY_ENABLED_UNIVERSES
+            and self._trust_boundary_enabled
```

Comment:
should we throw an error or log something if the user is trying to enable trust boundaries but isn't in a supported domain? Seems like we just ignore it silently

Reply:
To check the enable of lookup itself is a temporary solution. That's for injection test phase when backend is not ready on lookup endpoint. Will be totally removed. Then since it's an `and` condition, it won't be true anyway if we are not in GDU.